sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/PropertiesAccessCorrectField.cs

```diff
@@ -54,7 +54,9 @@ protected override IEnumerable<FieldData> FindFieldAssignments(IPropertySymbol p
                 FieldData? foundField = null;
                 if (node is AssignmentExpressionSyntax assignment && node.IsKind(SyntaxKind.SimpleAssignmentExpression))
                 {
-                    foundField = ExtractFieldFromExpression(AccessorKind.Setter, assignment.Left, compilation);
+                    foundField = assignment.Left.DescendantNodesAndSelf().OfType<ExpressionSyntax>()
+                        .Select(x => ExtractFieldFromExpression(AccessorKind.Setter, x, compilation))
```

Comment:
pedantic: "x" does not provide any helpful info, I think that using "expression" would provide a bit of context

Reply:
I don't want to change this. It's lambda, so single letter variable is fine. `x` is by convention the most used one (I've found 63 other files in project with `x =>`). 
It comes from `.OfType<ExpressionSyntax>()` so in this case, it's very clear and explicit what `x` is.